### PR TITLE
Fixed NumPy deprecations

### LINKED ISSUE #2122: replace deprecated np.matrix operations with operations on np.arrays
**Expected behavior**

No warnings regarding numpy use should be issued when using MDAnalysis.

**Actual behavior**

For example:
```
.../mdanalysis/package/MDAnalysis/analysis/align.py:288: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
```

**Code to reproduce the behavior**

```
pytest  MDAnalysisTests/analysis/test_align.py
```

**Currently version of MDAnalysis**

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 0.19.0-dev
- Which version of Python (`python -V`)? Python 3.6.6
- Which operating system? macOS

### LINKED ISSUE #2123: replace deprecated numpy sequence slices with tuple slices
**Expected behavior**

No warnings regarding numpy use should be issued when using MDAnalysis.

**Actual behavior**

For example:
```
... mdanalysis/package/MDAnalysis/coordinates/memory.py:485: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.
  ts._replace_positions_array(self.coordinate_array[basic_slice])
```

**Code to reproduce the behavior**

```
pytest  MDAnalysisTests/analysis/test_align.py
```

**Currently version of MDAnalysis**

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 0.19.0-dev
- Which version of Python (`python -V`)? Python 3.6.6
- Which operating system? macOS

### PR DESCRIPTION
**Pull request by PicoCentauri**
Fixes #2122

Fixes #2123 

Changes made in this Pull Request:
Fixed some NumPy depreciation warnings including those mentioned in issue #2122 and #2123.

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/rms.py [hunk 1]
```diff
@@ -587,7 +587,7 @@ def _prepare(self):
             #    (x~: selected coordinates, x: all coordinates)
             # Final transformed traj coordinates: x' = (x-x~_com)*R + ref_com
             self._rot = np.zeros(9, dtype=np.float64)  # allocate space
-            self._R = np.matrix(self._rot.reshape(3, 3))
+            self._R = np.array(self._rot.reshape(3, 3))
```
Comment (PicoCentauri) [inline package/MDAnalysis/analysis/rms.py]: Should we note the change in the docstring?

File: package/MDAnalysis/analysis/align.py [hunk 2]
```diff
@@ -285,7 +285,7 @@ def rotation_matrix(a, b, weights=None):
     # so that R acts **to the left** and can be broadcasted; we're saving
     # one transpose. [orbeckst])
     rmsd = qcp.CalcRMSDRotationalMatrix(a, b, N, rot, weights)
-    return np.matrix(rot.reshape(3, 3)), rmsd
+    return np.array(rot.reshape(3, 3)), rmsd
```
Comment (richardjgowers) [inline package/MDAnalysis/analysis/align.py]: if `rot` can do `.reshape` it's probably already an array? so just return `rot.reshape`

File: package/MDAnalysis/analysis/helanal.py [hunk 3]
```diff
@@ -749,15 +749,14 @@ def rotation_angle(helix_vector, axis_vector, rotation_vector):
 def vector_of_best_fit(origins):
     origins = np.asarray(origins)
     centroids = center(origins)
-    M = np.matrix(origins - centroids)
-    A = M.transpose() * M
+    M = np.array(origins - centroids)
```
Comment (richardjgowers) [inline package/MDAnalysis/analysis/helanal.py]: same here, probably don't need the `np.array` call if the subtraction already works

File: package/MDAnalysis/analysis/rms.py [hunk 4]
```diff
@@ -101,7 +101,7 @@
    ax.plot(time, rmsd[5], 'b--', label="NMP")
    ax.legend(loc="best")
    ax.set_xlabel("time (ps)")
-   ax.set_ylabel(r"RMSD ($\AA$)")
+   ax.set_ylabel(r"RMSD ($\\\\AA$)")
```
Comment (richardjgowers) [inline package/MDAnalysis/analysis/rms.py]: ?

File: package/MDAnalysis/analysis/rms.py (hunk 1 above)
Comment (richardjgowers) [reply package/MDAnalysis/analysis/rms.py]: I don't think so, this change should mostly be invisible

Not sure you need to `np.array` call

File: package/MDAnalysis/coordinates/base.py [hunk 5]
```diff
@@ -501,24 +501,23 @@ def copy_slice(self, sel):
            copy official attributes of the Timestep.
 
         """
-        # Detect the size of the Timestep by doing a dummy slice
```
Comment (richardjgowers) [inline package/MDAnalysis/coordinates/base.py]: need this comment

File: package/MDAnalysis/analysis/rms.py (hunk 1 above)
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/rms.py]: done

File: package/MDAnalysis/analysis/rms.py (hunk 4 above)
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/rms.py]: Only one backslash throws a warning, so two are needed. 4 is definitely two much ;).

File: package/MDAnalysis/analysis/align.py (hunk 2 above)
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/align.py]: done

File: package/MDAnalysis/analysis/helanal.py (hunk 3 above)
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/helanal.py]: done

File: package/MDAnalysis/coordinates/base.py (hunk 5 above)
Comment (PicoCentauri) [reply package/MDAnalysis/coordinates/base.py]: done

**Comment by richardjgowers** [comment]
Thanks @PicoCentauri another one down!

**Pull request merged**